Adopt smart pointers to glib related code

mikhailramalho opened the pull request:
#### fa36680627bf7c079a0d1f1b88b171889342af25
<pre>
Adopt smart pointers to glib related code
<a href="https://bugs.webkit.org/show_bug.cgi?id=274898">https://bugs.webkit.org/show_bug.cgi?id=274898</a>

Reviewed by Carlos Garcia Campos.

This patch changes glib related code in WebKit to
use smart pointers

* Source/JavaScriptCore/API/glib/JSCClass.cpp:
(getProperty):
(setProperty):
(hasProperty):
(deleteProperty):
(getPropertyNames):
(jscClassCreate):
(jscClassCreateConstructor):
(jscClassAddMethod):
* Source/JavaScriptCore/API/glib/JSCContext.cpp:
(jscContextPushCallback):
(jscContextPopCallback):
(jscContextGarbageCollect):
(jsc_context_evaluate_in_object):
(jsc_context_check_syntax):
* Source/JavaScriptCore/API/glib/JSCException.cpp:
(jscExceptionCreate):
* Source/JavaScriptCore/API/glib/JSCValue.cpp:
(jsc_value_object_enumerate_properties):
(jsc_value_object_define_property_data):
(jscValueObjectDefinePropertyAccessor):
(jscValueFunctionCreate):
(jsc_value_is_array_buffer):
(jsc_value_typed_array_get_type):
* Source/JavaScriptCore/API/glib/JSCWeakValue.cpp:
(jsc_weak_value_get_value):
* Source/JavaScriptCore/API/glib/JSCWrapperMap.cpp:
(JSC::WrapperMap::registerClass):
(JSC::WrapperMap::createJSWrapper):
* Source/JavaScriptCore/inspector/remote/glib/RemoteInspectorGlib.cpp:
(Inspector::RemoteInspector::sendMessageToTarget):
* Source/JavaScriptCore/inspector/remote/glib/RemoteInspectorServer.cpp:
(Inspector::RemoteInspectorServer::setTargetList):
(Inspector::RemoteInspectorServer::setupInspectorClient):
(Inspector::RemoteInspectorServer::connectionDidClose):
(Inspector::RemoteInspectorServer::sendMessageToFrontend):
* Source/WTF/wtf/glib/RunLoopGLib.cpp:
(WTF::RunLoop::run):
* Source/WebKit/UIProcess/API/glib/WebKitAutomationSession.cpp:
(webkitAutomationSessionCreate):
* Source/WebKit/UIProcess/API/glib/WebKitBackForwardList.cpp:
(webkitBackForwardListCreateList):
* Source/WebKit/UIProcess/API/glib/WebKitCookieManager.cpp:
(webkit_cookie_manager_set_persistent_storage):
(webkit_cookie_manager_set_accept_policy):
* Source/WebKit/UIProcess/API/glib/WebKitFileChooserRequest.cpp:
(webkit_file_chooser_request_get_mime_types):
(webkit_file_chooser_request_get_selected_files):
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSession.cpp:
(webkit_network_session_set_itp_enabled):
(webkit_network_session_get_itp_enabled):
(webkit_network_session_set_persistent_credential_storage_enabled):
(webkit_network_session_get_persistent_credential_storage_enabled):
(webkit_network_session_set_tls_errors_policy):
(webkit_network_session_allow_tls_certificate_for_host):
(webkit_network_session_set_proxy_settings):
(webkit_network_session_get_itp_summary):
(webkit_network_session_prefetch_dns):
(webkit_network_session_download_uri):
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/API/glib/WebKitSecurityManager.cpp:
(registerSecurityPolicyForURIScheme):
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webKitSettingsConstructed):
* Source/WebKit/UIProcess/API/glib/WebKitUIClient.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebResourceLoadManager.cpp:
(WebKit::WebKitWebResourceLoadManager::didInitiateLoad):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(zoomTextOnlyChanged):
(webkitWebViewUpdateSettings):
(webkit_web_view_set_zoom_level):
(webkit_web_view_get_zoom_level):
* Source/WebKit/UIProcess/glib/WebProcessProxyGLib.cpp:
(WebKit::WebProcessProxy::platformGetLaunchOptions):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.cpp:
(getURL):
(webkitFrameGetJSCValuesForElementsInWorld):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManager.cpp:
(webkit_web_form_manager_input_element_is_user_edited):
(webkit_web_form_manager_input_element_auto_fill):
(webkit_web_form_manager_input_element_is_auto_filled):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebHitTestResult.cpp:
(webkit_web_hit_test_result_get_js_node):
* Source/WebKit/WebProcess/WebCoreSupport/glib/WebEditorClientGLib.cpp:
(WebKit::WebEditorClient::didDispatchInputMethodKeydown):
* Source/WebKit/WebProcess/WebPage/glib/WebPageGLib.cpp:
(WebKit::WebPage::platformInitialize):

Canonical link: <a href="https://commits.webkit.org/280015@main">https://commits.webkit.org/280015@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d6757505bb30473a65782ce1bf2bdd4a7421792b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55071 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34538 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7686 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58353 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5803 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42096 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5842 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44587 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3963 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57166 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32613 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47712 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25711 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29393 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5059 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3947 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/48446 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51462 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5294 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59943 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/54576 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30350 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5460 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52017 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31474 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47780 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51470 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12298 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32500 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/66872 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31264 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12747 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/JavaScriptCore/API/glib/JSCClass.cpp:
```diff
@@ -477,9 +477,9 @@ GRefPtr<JSCClass> jscClassCreate(JSCContext* context, const char* name, JSCClass
-    JSClassRef prototypeClass = JSClassCreate(&prototypeDefinition);
-    priv->prototype = jscContextGetOrCreateJSWrapper(context, prototypeClass);
-    JSClassRelease(prototypeClass);
+    RefPtr prototypeClass = JSClassCreate(&prototypeDefinition);
+    priv->prototype = jscContextGetOrCreateJSWrapper(context, prototypeClass.get());
+    JSClassRelease(prototypeClass.get());
```
Don't release it manually. Use adoptRef().

carlosgcampos replied:
JSClassRef is not a GObject, but a C API object. so I guess you mean to use JSRetainPtr

carlosgcampos replied:
Ah!, I see you wrote adoptRef, not adoptGRef, so it's indeed right

mikhailramalho committed:
Adopt smart pointers to glib related code
https://bugs.webkit.org/show_bug.cgi?id=274898

Reviewed by Carlos Garcia Campos.

This patch changes glib related code in WebKit to
use smart pointers

* Source/JavaScriptCore/API/glib/JSCClass.cpp:
(getProperty):
(setProperty):
(hasProperty):
(deleteProperty):
(getPropertyNames):
(jscClassCreate):
(jscClassCreateConstructor):
(jscClassAddMethod):
* Source/JavaScriptCore/API/glib/JSCContext.cpp:
(jscContextPushCallback):
(jscContextPopCallback):
(jscContextGarbageCollect):
(jsc_context_evaluate_in_object):
(jsc_context_check_syntax):
* Source/JavaScriptCore/API/glib/JSCException.cpp:
(jscExceptionCreate):
* Source/JavaScriptCore/API/glib/JSCValue.cpp:
(jsc_value_object_enumerate_properties):
(jsc_value_object_define_property_data):
(jscValueObjectDefinePropertyAccessor):
(jscValueFunctionCreate):
(jsc_value_is_array_buffer):
(jsc_value_typed_array_get_type):
* Source/JavaScriptCore/API/glib/JSCWeakValue.cpp:
(jsc_weak_value_get_value):
* Source/JavaScriptCore/API/glib/JSCWrapperMap.cpp:
(JSC::WrapperMap::registerClass):
(JSC::WrapperMap::createJSWrapper):
* Source/JavaScriptCore/inspector/remote/glib/RemoteInspectorGlib.cpp:
(Inspector::RemoteInspector::sendMessageToTarget):
* Source/JavaScriptCore/inspector/remote/glib/RemoteInspectorServer.cpp:
(Inspector::RemoteInspectorServer::setTargetList):
(Inspector::RemoteInspectorServer::setupInspectorClient):
(Inspector::RemoteInspectorServer::connectionDidClose):
(Inspector::RemoteInspectorServer::sendMessageToFrontend):
* Source/WTF/wtf/glib/RunLoopGLib.cpp:
(WTF::RunLoop::run):
* Source/WebKit/UIProcess/API/glib/WebKitAutomationSession.cpp:
(webkitAutomationSessionCreate):
* Source/WebKit/UIProcess/API/glib/WebKitBackForwardList.cpp:
(webkitBackForwardListCreateList):
* Source/WebKit/UIProcess/API/glib/WebKitCookieManager.cpp:
(webkit_cookie_manager_set_persistent_storage):
(webkit_cookie_manager_set_accept_policy):
* Source/WebKit/UIProcess/API/glib/WebKitFileChooserRequest.cpp:
(webkit_file_chooser_request_get_mime_types):
(webkit_file_chooser_request_get_selected_files):
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSession.cpp:
(webkit_network_session_set_itp_enabled):
(webkit_network_session_get_itp_enabled):
(webkit_network_session_set_persistent_credential_storage_enabled):
(webkit_network_session_get_persistent_credential_storage_enabled):
(webkit_network_session_set_tls_errors_policy):
(webkit_network_session_allow_tls_certificate_for_host):
(webkit_network_session_set_proxy_settings):
(webkit_network_session_get_itp_summary):
(webkit_network_session_prefetch_dns):
(webkit_network_session_download_uri):
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/API/glib/WebKitSecurityManager.cpp:
(registerSecurityPolicyForURIScheme):
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webKitSettingsConstructed):
* Source/WebKit/UIProcess/API/glib/WebKitUIClient.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebResourceLoadManager.cpp:
(WebKit::WebKitWebResourceLoadManager::didInitiateLoad):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(zoomTextOnlyChanged):
(webkitWebViewUpdateSettings):
(webkit_web_view_set_zoom_level):
(webkit_web_view_get_zoom_level):
* Source/WebKit/UIProcess/glib/WebProcessProxyGLib.cpp:
(WebKit::WebProcessProxy::platformGetLaunchOptions):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.cpp:
(getURL):
(webkitFrameGetJSCValuesForElementsInWorld):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManager.cpp:
(webkit_web_form_manager_input_element_is_user_edited):
(webkit_web_form_manager_input_element_auto_fill):
(webkit_web_form_manager_input_element_is_auto_filled):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebHitTestResult.cpp:
(webkit_web_hit_test_result_get_js_node):
* Source/WebKit/WebProcess/WebCoreSupport/glib/WebEditorClientGLib.cpp:
(WebKit::WebEditorClient::didDispatchInputMethodKeydown):
* Source/WebKit/WebProcess/WebPage/glib/WebPageGLib.cpp:
(WebKit::WebPage::platformInitialize):

Canonical link: https://commits.webkit.org/280015@main
